examples/direct_fidelity_estimation.py

```diff
@@ -0,0 +1,82 @@
+"""Implements direct fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  # Builds an arbitrary circuit to test. The circuit is non Clifford to show the
+  # use of simulators.
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def compute_characteristic_function(circuit, P_i, qubits, noise):
```

Comment:
Thanks for adding the comments so I know what you're trying to compute :)

In the linked paper, rho_i and sigma_i refer to Tr[rho P_i] or Tr[sigma P_i], resp. where rho and/or sigma are density matrices and the P_i are Hermitian operators (in this case, `PauliString`s) 

The secret is to realize that Tr[rho P_i] is the expectation value of P_i given the state rho. There is a function that does this for you in cirq! `PauliString.expectation_value_from_density_matrix`. 

Of course, to certify a state from the actual quantum computer we don't have access to the density matrix so we can talk about how to slot in a sampling-based estimate for operator estimation values to be implemented in a follow-up PR